x-pack/plugins/ml/public/components/kql_filter_bar/utils.js

```diff
@@ -88,9 +88,15 @@ export function removeFilterFromQueryString(currentQueryString, fieldName, field
   // match 'and' or 'or' at the start/end of the string
   const endPattern = /\s(and|or)\s*$/ig;
   const startPattern = /^\s*(and|or)\s/ig;
+  // If string has a double 'and' operator (e.g. tag:thing and and tag:other) remove as it is illegal kuery syntax
+  const invalidAndPattern = /\s+(and)\s+(and)\s+/ig;
+  // If string has a double 'or' operator (e.g. tag:thing or or tag:other) remove as it is illegal kuery syntax
+  const invalidOrPattern = /\s+(or)\s+(or)\s+/ig;
```

Comment:
this won't catch ` OR AND ` or ` AND OR `.
the regex could be changed to something like `/\s+(and|or)\s+(and|or)\s+/ig` to catch all combinations of duplicate operators.